controllers/cluster_controller.go

```diff
@@ -149,8 +143,6 @@ func (r *FoundationDBClusterReconciler) Reconcile(ctx context.Context, request c
 	}
 
 	for _, subReconciler := range subReconcilers {
-		cluster.Spec = *(normalizedSpec.DeepCopy())
```

Comment:
It seems like we need that 🤔

Reply:
It's probably worth looking into why we need this, since I don't remember and I think I'm the one who wrote this line. It's possible we have subreconcilers that are fetching the cluster again and overwriting the normalized spec with a non-normalized spec. That seems like a bad practice that could cause trouble down the line regardless.

Reply:
Yes, that's a good point. I need to spent some time fixing the tests and se why we needed that.

Reply:
After spending some time debugging that I figured out that the controller runtime will reset these fields when we do an update to the status (I assume that it just takes the latest version):

```bash
{"level":"info","ts":1620813027.731017,"logger":"controller","msg":"Update Status Before!","pod limits":{"cpu":"250m","memory":"128Mi"}}
{"level":"info","ts":1620813027.748231,"logger":"controller","msg":"Update Status After!","pod limits":null}
```

That's pretty annoying but I'm not sure of any better workaround than setting the spec again :/